### PR TITLE
UnsubscribeAll Event

### PR DESCRIPTION
**Pull request by budgetpreneur**

### CODE REVIEW

**Comment by mohitpubnub** [comment]
Is it necessary to have channels and channelGroups value with UnsubscribeAll Event?
Isn't it the event type itself suggesting to remove all channels and groups from subscription.

**Review by MikeDobrzan — CHANGES_REQUESTED** [review]
UnsubscribeAll event and Unsubscribed state does not need any information about channels and channel groups

File: src/Api/PubnubApi/EventEngine/Subscribe/States/UnsubscribedState.cs [hunk 1]
```diff
@@ -7,6 +7,8 @@ namespace PubnubApi.PubnubEventEngine.Subscribe.States
 {
     internal class UnsubscribedState : Core.IState
     {
+        public IEnumerable<string> Channels;
+        public IEnumerable<string> ChannelGroups;
```
[the remark this answers is not in the document]
Comment (budgetpreneur) [reply src/Api/PubnubApi/EventEngine/Subscribe/States/UnsubscribedState.cs:11]: I added `LeaveEffectHandler` where `LeaveInvocation` needs channels and channel groups to make REST API leave call through SubscribeManager2.
If we are doing different way, we can discuss on it.

[the remark this answers is not in the document]
Comment (budgetpreneur) [reply src/Api/PubnubApi/EventEngine/Subscribe/States/UnsubscribedState.cs]: Rolled back Leave stuff.
Removed channels and channel groups from UnsubscribeAll.

File: src/Api/PubnubApi/EventEngine/Subscribe/Events/SubscriptionEvents.cs [hunk 2]
```diff
@@ -1,7 +1,11 @@
 ﻿using System.Collections.Generic;
 
 namespace PubnubApi.PubnubEventEngine.Subscribe.Events {
-	public class SubscriptionChangedEvent : Core.IEvent {
+    public class UnsubscribeAllEvent : Core.IEvent {
+		public IEnumerable<string> Channels;
+		public IEnumerable<string> ChannelGroups;
+    }
```
[the remark this answers is not in the document]
Comment (budgetpreneur) [reply src/Api/PubnubApi/EventEngine/Subscribe/Events/SubscriptionEvents.cs:5]: ok..removed them.

**Comment by budgetpreneur** [comment]
> channels and groups

removed  channels and groups

**Review by mohitpubnub — APPROVED** [review]

**Pull request merged**